plugins/filter_aws/aws.c

```diff
@@ -427,8 +429,135 @@ static int get_ec2_tag_values(struct flb_filter_aws *ctx)
     return 0;
 }
 
+static int tag_is_present_in_list(struct flb_filter_aws *ctx, flb_sds_t tag,
+        flb_sds_t *tags, int tags_n)
+{
+    int i;
+    for (i = 0; i < tags_n; i++) {
+        if (strcmp(tag, tags[i]) == 0) {
+            return FLB_TRUE;
+        }
+    }
+    return FLB_FALSE;
+}
+
+static int tags_split(char *tags, flb_sds_t **tags_list, int *tags_list_n) {
+    flb_sds_t token;
+    int i;
+    int n;
+    n = 1;
+    for (i = 0; i < strlen(tags); i++) {
+        if (tags[i] == ',') {
+            n++;
+        }
+    }
+
+    *tags_list = calloc(sizeof(flb_sds_t), n);
+    if (*tags_list == NULL) {
+        return -2;
+    }
+
+    token = strtok(tags, ",");
+    i = 0;
+    while (token != NULL) {
+        (*tags_list)[i] = token;
+        i++;
+        token = strtok(NULL, ",");
+    }
+
+    *tags_list_n = n;
+
+    return 0;
+}
+
+static int get_ec2_tag_enabled(struct flb_filter_aws *ctx)
+{
+    const char *tags_include;
+    const char *tags_exclude;
+    char *tags_copy;
+    flb_sds_t *tags;
+    int tags_n;
+    int i;
+    int tag_present;
+    int result;
+
+    /* if there are no tags, there is no need to evaluate which tag is enabled */
+    if (ctx->tags_count == 0) {
+        return 0;
+    }
+
+
+    /* allocate memory for 'tag_is_enabled' for all tags */
+    ctx->tag_is_enabled = flb_calloc(ctx->tags_count, sizeof(int));
+    if (!ctx->tag_is_enabled) {
+        flb_plg_error(ctx->ins, "Failed to allocate memory for tag_is_enabled");
+        return -1;
+    }
+
+    /* if tags_include and tags_exclude are not defined, set all tags as enabled */
+    for (i = 0; i < ctx->tags_count; i++) {
+        ctx->tag_is_enabled[i] = FLB_TRUE;
+    }
+
+    /* apply tags_included configuration */
+    tags_include = flb_filter_get_property("tags_include", ctx->ins);
+    if (tags_include) {
+        /* copy const string in order to use strtok which modifes the string */
+        tags_copy = strdup(tags_include);
```

Comment:
use `flb_strndup` or `flb_strdup` in `flb_str.h`.

Reply:
I changed the function call to `flb_strdup` in https://github.com/fluent/fluent-bit/pull/6866/commits/b568f346d0ea09de18fe280cd2b004fd7c777708